### DISCUSSION TITLE
Is SpaCy compatible with CUDA 11.4 and is Pytorch required

### DISCUSSION THREAD

**Discussion by XavBeckers** [explosion/spaCy#9268] [Help: Installation] [answered]
Hello,

I'm trying (with a huge lot of troubles) to setup spaCy to be able to train models on my GPU (RTX 3060 Ti). As i see using nvidia-smi, my driver is compatible with CUDA 11.4. As it is easier to install CUDA 11.4 using Nvidia's instructions, i'd like to know if it is possible to use spaCy with CUDA 11.4 installed considering this version it is not available in the installation instructions.

Moreover, i'd like to know if it is necessary to have pytorch working beforehand installing spacy.  As my GPU has the Ampere architecture, it doesn't work at first with pytorch and i have to update pytorch myself to get it working. The problem is pytorch then installs CUDA 11.1 and i fear it could mess up my whole installation. 

I'm working on a brand new Ubuntu 20.04 environment and i'm kinda new to linux. 

Thank you in advance.

**Comment by svlandeg** [comment, marked as answer]
Hi!

You can use spaCy without PyTorch, unless you want to run/train transformers models. 

If you want to use transformers, you'd need to install `spacy-transformers` which relies on PyTorch.

I've found that it's typically best to install things in this order:

- Install a CUDA version that is compatible with PyTorch (their [quickstart](https://pytorch.org/get-started/locally/) seems to indicate that 11.1 is the highest support one)
- Install PyTorch with the command generated from their quickstart
- `pip install spacy`
- `pip install cupy-cuda111` (or a different version according to your first step)
You can also do `pip install spacy[cuda111]` as a shorthand for those last two steps.
- `pip install spacy-transformers`

If you don't want to work with transformer models and don't want to install PyTorch, I think you can try setting things up for CUDA 11.4:
```
pip install cupy-cuda114
pip install spacy
```
I think theoretically that should work - let us know if it doesn't!

**Reply by svlandeg** [reply]
PS: note that the shorthand `pip install spacy[cudaXXX]` currently only works up untill CUDA 11.2 - we should probably extend that further even if PyTorch doesn't support a higher version yet, but it would require us to test things first on CUDA 11.3 and 11.4. See also https://spacy.io/usage#gpu.

**Reply by XavBeckers** [reply]
Thanks a lot! I'll try like this with CUDA 11.1 as I need transformers.

**Reply by felipefoschiera** [reply]
Hi @XavBeckers , did you manage to get it working? I tried all of the steps above on Ubuntu WSL for my RTX 3060 Ti. The commands such as `torch.cuda.is_available()` and `spacy.require_gpu()` both work, but when I try to train a transformer model I get:
`CUDA error: no kernel image is available for execution on the device`.
